File: src/backend/notification-service/Models/Entities/Shop.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using TenantlyNotificationService.Models.Interfaces;

namespace TenantlyNotificationService.Models.Entities;

[Table("shops")]
public class Shop : IAuditableEntity
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    [Column("shop_number")]
    public string ShopNumber { get; set; } = string.Empty;

    [MaxLength(10)]
    [Column("floor")]
    public string? Floor { get; set; }

    [MaxLength(50)]
    [Column("section")]
    public string? Section { get; set; }

    [Required]
    [Column("monthly_rent", TypeName = "decimal(10,2)")]
    public decimal MonthlyRent { get; set; }

    [Column("active")]
    public bool Active { get; set; } = true;

    [Column("property_id")]
    public int PropertyId { get; set; } = 1;

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    [ForeignKey("PropertyId")]
    public virtual Property Property { get; set; } = null!;
    
    public virtual ICollection<Lease> Leases { get; set; } = new List<Lease>();
    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();
    public virtual ICollection<NotificationQueue> Notifications { get; set; } = new List<NotificationQueue>();
    public virtual ICollection<Attachment> Attachments { get; set; } = new List<Attachment>();
}

File: src/backend/api/internal/models/shop.go
package models

import "time"

type Shop struct {
	ID          int       `json:"id" db:"id"`
	Name        string    `json:"name" db:"name"`
	ShopNumber  string    `json:"shop_number" db:"shop_number"`
	Floor       string    `json:"floor" db:"floor"`
	Section     string    `json:"section" db:"section"`
	MonthlyRent float64   `json:"monthly_rent" db:"monthly_rent"`
	Active      bool      `json:"active" db:"active"`
	PropertyID  int       `json:"property_id" db:"property_id"`
	CreatedAt   time.Time `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time `json:"updated_at" db:"updated_at"`
}

type CreateShopRequest struct {
	Name        string  `json:"name" binding:"required,max=100"`
	ShopNumber  string  `json:"shop_number" binding:"required,max=20"`
	Floor       string  `json:"floor" binding:"omitempty,max=10"`
	Section     string  `json:"section" binding:"omitempty,max=50"`
	MonthlyRent float64 `json:"monthly_rent" binding:"required,gt=0"`
	PropertyID  int     `json:"property_id" binding:"omitempty"`
}

type UpdateShopRequest struct {
	Name        string   `json:"name" binding:"omitempty,max=100"`
	ShopNumber  string   `json:"shop_number" binding:"omitempty,max=20"`
	Floor       string   `json:"floor" binding:"omitempty,max=10"`
	Section     string   `json:"section" binding:"omitempty,max=50"`
	MonthlyRent *float64 `json:"monthly_rent" binding:"omitempty,gt=0"`
	Active      *bool    `json:"active"`
}
